examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,491 @@
+)
+coefs = pd.DataFrame(
+    [est.named_steps['transformedtargetregressor'].regressor_.coef_
+     for est in cv_model['estimator']],
+    columns=feature_names
+)
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefs, orient='h', color='k', alpha=0.5)
+sns.boxplot(data=coefs, orient='h', color='cyan', saturation=0.5)
+plt.axvline(x=0, color='.5')
+plt.title('Coefficient variability')
+plt.subplots_adjust(left=.3)
+
+##############################################################################
+# The result is quite similar to the non-normalised case.
+#
+# Linear models with regularization
+# ---------------------------------
+#
+# In practice, Ridge Regression is more often used with some regularization.
+# Regularization improves the conditioning of the problem and reduces the
+# variance of the estimates. RidgeCV applies cross validation in order to
+# determine which value of the regularization parameter (`alpha`) is best
+# suited for the model estimation.
+
+from sklearn.linear_model import RidgeCV
+from sklearn.compose import TransformedTargetRegressor
+
+model = make_pipeline(
+    preprocessor,
+    TransformedTargetRegressor(
+        regressor=RidgeCV(alphas=np.logspace(-10, 10, 21)),
+        func=np.log10,
+        inverse_func=sp.special.exp10
+    )
+)
+
+model.fit(X_train, y_train)
+
+##############################################################################
+# First we verify which value of :math:`\alpha` has been selected.
+
+model[-1].regressor_.alpha_
+
+##############################################################################
+# Then we check the quality of the predictions.
+
+y_pred = model.predict(X_train)
+mae = median_absolute_error(y_train, y_pred)
+string_score = 'MAE on training set: {0:.2f} $/hour'.format(mae)
+y_pred = model.predict(X_test)
+mae = median_absolute_error(y_test, y_pred)
+r2score = model.score(X_test, y_test)
+
+string_score += '\nMAE on testing set: {0:.2f} $/hour'.format(mae)
+string_score += '\nR2 score: {0:.4f}'.format(r2score)
+fig, ax = plt.subplots(figsize=(6, 6))
+sns.regplot(y_test, y_pred)
+
+plt.text(3, 20, string_score)
+
+plt.ylabel('Model predictions')
+plt.xlabel('Truths')
+plt.xlim([0, 27])
+plt.ylim([0, 27])
+
+##############################################################################
+# The R squared coefficient is similar to the non-regularized case.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_,
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+##############################################################################
+# Coefficients are significantly different.
+# AGE and EXPERIENCE coefficients are both positive.
+# Even if the model is still not able to provide a good description of the
+# dataset, the regularization manages to lower the influence of correlated
+# variables on the model.
```

Comment:
I'm somewhat disappointed that the comparison with Lasso did not make it to here :). I'll leave it as an open question on whether or not we should address it.

Reply:
I will do some tests in the [notebook](https://github.com/cmarmo/linearcoeff_interp), and ping you when it's done.

Reply:
@GaelVaroquaux LassoCV implementation is available at the end of the [notebook](https://github.com/cmarmo/linearcoeff_interp): Lasso choose between AGE and EXPERIENCE... but it seems to me that the other coefficients are a bit biased....

Reply:
Addressed in the example. My discussion is quite reduced, feel free to add more insight.